### ISSUE TITLE
Support Use Alongside chai-as-promised

### ISSUE THREAD

**Issue by lobobabysaurus** [electricmonk/sinon-chai-in-order#2]
When sinon-chai-in-order is used as a chai plugin, [chai-as-promised](https://github.com/domenic/chai-as-promised) begins to show a host of problems.  Particularly I have seen consistent `TypeError: Cannot set property then of [object Object] which has only a getter` and `RangeError: Maximum call stack size exceeded` errors.

I'm not sure how much can be done or how much coordination would need to happen between the two plugin projects, but it is a real pain to have to pick between which super useful plugin to use.

Here is some example code in ES6 which allows reproduction of the issue and requires chai, chai-as-promised and sinon-chai-in-order to be installed

```
import chai from 'chai';
import sinonChaiInOrder from 'sinon-chai-in-order';
import chaiAsPromised from 'chai-as-promised';

chai.should();
chai.use(chaiAsPromised);
chai.use(sinonChaiInOrder);

describe('chai as promised', () => {
  it('should properly handle es6 promises', () => {
    Promise.resolve(2+2).should.eventually.equal(4);
  })
});
```

If they cant be used simultaneously I think the minimum that should happen should be a note in the README which warns people about the compatibility issue to hopefully save some time and frustration.

Same issue has been opened with chai-as-promised [here](https://github.com/domenic/chai-as-promised/issues/140)

**Comment by electricmonk** [comment]
Seems like a bug in `sinon-chai-in-order`, where it assumes that `assertion` doesn't already have a `then` function. A temporary solution would be to override and cancel its default behavior:

``` js
chaiAsPromised.transferPromiseness = function(a, b) {};
```

**Comment by lobobabysaurus** [comment]
Solid, To be able to support them both simultaneously I'll probably fork this repo and change line 13 of sinon-chai-in-order.js to be something like 'subsequently' instead of 'then' so that there is less chance of it conflicting with the .then function that most promises have.  Nothing really feels as right as using 'then' in this project, but I feel like with most promise libraries using 'then' as the default handling method, it opens itself up for conflicts.

**Comment by electricmonk** [comment]
@lobobabysaurus, we can rename `then` to `followedBy` or `subsequently` (or both!) and bump the major version. Feel like sending me a PR?

**Comment by lobobabysaurus** [comment]
Have the PR here - https://github.com/electricmonk/sinon-chai-in-order/pull/3. Updated the README, src, extant tests, and added a new test.  Didn't touch the versioning.

**Comment by electricmonk** [comment]
Released [0.1.0](https://github.com/electricmonk/sinon-chai-in-order/releases/tag/0.1.0)
